src/framework/mlt_repository.c

```diff
@@ -98,12 +98,51 @@ mlt_repository mlt_repository_init( const char *directory )
 	free(newpath);
 #endif
 
+	char **denylist  = NULL;
+	int dl_length = 0;
+	if (getenv( "MLT_REPOSITORY_DENY" ))
+	{
+		char *p = strtok (getenv( "MLT_REPOSITORY_DENY" ), ":");
+
+		while (p) {
+			denylist = realloc (denylist, sizeof (char*) * ++dl_length);
+
+			// check if memory allocation failed
+			if (denylist == NULL) {	break; }
+
+			denylist[dl_length-1] = p;
+			p = strtok (NULL, ":");
+		}
+
+		// realloc one extra element for the last NULL
+		denylist = realloc (denylist, sizeof (char*) * (dl_length + 1));
+		denylist[dl_length] = 0;
+	}
+
 	// Iterate over files
 	for ( i = 0; i < count; i++ )
 	{
 		int flags = RTLD_NOW;
 		const char *object_name = mlt_properties_get_value( dir, i);
 
+		// check if the plugin was asked to be skipped through MLT_REPOSITORY_DENY
+		int ignore = 0;
+		for (int j = 0; j < dl_length; ++j)
+		{
+			char *denyfile = calloc( 1, strlen( directory ) + strlen( denylist[j] ) + strlen( LIBSUF ) + 1 );
```

Comment:
I prefer not to consider explicit extension because on macOS it needs to allow either .so or .dylib. Rather it should consider basename minus the extension. In the case of multiple multiple periods in the basename the first one should be considered the start of the extension.

Reply:
Changed